src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -199,13 +200,31 @@ def __init__(
             str, Dict[str, str]
         ] = defaultdict(dict)
         self._api_calls = api_calls.copy() if api_calls is not None else None
+        self.source_plan = self
+
```

Comment:
if `self._snowflake_plan` is not `None`, should this be `self._snowflake_plan.source_plan`?

Reply:
source_plan` here is just a shortcut can help us run CTE algorithm (it requires `source_plan` https://github.com/snowflakedb/snowpark-python/blob/dd27f6e9b12d5decc2785d3cf7202d91f8b4215b/src/snowflake/snowpark/_internal/analyzer/cte_utils.py#L50) because the tree structure is maintained in source_plan of SnowflakePlan. Actually we can just use `node.children` if node is a Selectable. Therefore, we can only set `source_plan` to `self._snowflake_plan.source_plan` when it's a SelectSnowflakePlan (i.e., when Selectable is built from a SnowflakePlan)

Reply:
I left a suggestion on the updated commit to use `self.snowflake_plan.source_plan` which returns `source_plan` as `self` when `self._snowflake_plan` is None, and `self._snowflake_plan.source_plan` when it is non-None